tests/integration.rs

```diff
@@ -1,314 +1,602 @@
 #[cfg(test)]
 mod integration_tests {
     use anyhow::{Context, Result};
-    use hyper::{header::HeaderName, Body, Client, Response};
+    use hyper::{Body, Client, Response};
     use spin_loader::local::{
         config::{RawAppManifestAnyVersion, RawModuleSource},
         raw_manifest_from_file,
     };
     use std::{
         ffi::OsStr,
         net::{Ipv4Addr, SocketAddrV4, TcpListener},
-        path::{Path, PathBuf},
+        path::Path,
         process::{self, Child, Command},
         time::Duration,
     };
-    use tempfile::TempDir;
     use tokio::{net::TcpStream, time::sleep};
-    use which::which;
 
     const RUST_HTTP_INTEGRATION_TEST: &str = "tests/http/simple-spin-rust";
-    const RUST_HTTP_INTEGRATION_TEST_REF: &str = "spin-hello-world/1.0.0";
-
     const RUST_HTTP_STATIC_ASSETS_TEST: &str = "tests/http/assets-test";
-    const RUST_HTTP_STATIC_ASSETS_REST_REF: &str = "spin-assets-test/1.0.0";
-
-    const RUST_HTTP_HEADERS_ENV_ROUTES_TEST: &str = "tests/http/headers-env-routes-test";
-    const RUST_HTTP_HEADERS_ENV_ROUTES_TEST_REF: &str = "spin-headers-env-routes-test/1.0.0";
 
     const DEFAULT_MANIFEST_LOCATION: &str = "spin.toml";
 
     const SPIN_BINARY: &str = "./target/debug/spin";
-    const BINDLE_SERVER_BINARY: &str = "bindle-server";
-    const NOMAD_BINARY: &str = "nomad";
-    const HIPPO_BINARY: &str = "Hippo.Web";
 
-    const BINDLE_SERVER_PATH_ENV: &str = "SPIN_TEST_BINDLE_SERVER_PATH";
-    const BINDLE_SERVER_BASIC_AUTH_HTPASSWD_FILE: &str = "tests/http/htpasswd";
-    const BINDLE_SERVER_BASIC_AUTH_USER: &str = "bindle-user";
-    const BINDLE_SERVER_BASIC_AUTH_PASSWORD: &str = "topsecret";
+    // This module consist of all integration tests that require dependencies such as bindle-server, nomad, and Hippo.Web to be installed.
+    #[cfg(feature = "e2e-tests")]
+    mod e2e_tests {
```

Comment:
Would it make sense to break these out into a separate file?

Reply:
I chose to use a nested module since they are still integration tests and from my understanding our goal is for this to be a temporary gating until we have a setup were the dependencies can be installed at the beginning of integration testing (maybe as in the Fermyon installer): https://github.com/fermyon/spin/issues/626. Open to separating out for cleanliness if that is more desired.

Reply:
There is also a lot of shared structures (`SpinTestController`) and helper functions (`assert_status` etc) that the e2e_tests need that would need to be exported

Reply:
Okey dokey :+1:.

> from my understanding our goal is for this to be a temporary gating until we have a setup were the dependencies can be installed at the beginning of integration testing

I think we'll always want to keep tests that require external infrastructure separate from self-contained integration tests. We've even discussed splitting them out into a separate repo, but I think that is out of scope for this PR.